django/db/backends/sqlite3/operations.py

```diff
@@ -153,22 +151,30 @@ def get_db_converters(self, expression):
             converters.append(self.convert_uuidfield_value)
         return converters
 
-    def convert_decimalfield_value(self, value, expression, connection, context):
-        return backend_utils.typecast_decimal(expression.output_field.format_number(value))
+    def convert_datetimefield_value(self, value, expression, connection, context):
+        if value is not None:
+            if not isinstance(value, datetime.datetime):
+                value = parse_datetime(value)
+            if settings.USE_TZ:
+                value = timezone.make_aware(value, self.connection.timezone)
+        return value
 
     def convert_datefield_value(self, value, expression, connection, context):
-        if value is not None and not isinstance(value, datetime.date):
-            value = parse_date(value)
+        if value is not None:
+            if not isinstance(value, datetime.date):
+                value = parse_date(value)
```

Comment:
As explained above, there are about 20 `convert_*_value` in the code base, 15 of which used:

    if value is not None:
        if blablabla:
            ...

and 5 of which used:


    if value is not None and blablabla:
        ...

I normalized towards the most commonly used pattern.